Use more smart pointers in editing node

cdumez opened the pull request:
#### 9318f2ff1342ea508beeba6bf16691be3c06d65d
<pre>
Use more smart pointers in editing node
<a href="https://bugs.webkit.org/show_bug.cgi?id=261572">https://bugs.webkit.org/show_bug.cgi?id=261572</a>

Reviewed by Darin Adler and Brent Fulgham.

* Source/WebCore/dom/DocumentMarker.h:
(WebCore::DocumentMarker::DocumentMarker):
* Source/WebCore/dom/DocumentMarkerController.cpp:
(WebCore::DocumentMarkerController::markersFor):
(WebCore::DocumentMarkerController::markersInRange):
(): Deleted.
* Source/WebCore/dom/DocumentMarkerController.h:
* Source/WebCore/editing/AlternativeTextController.cpp:
(WebCore::markersHaveIdenticalDescription):
(WebCore::AlternativeTextController::respondToChangedSelection):
(WebCore::AlternativeTextController::recordSpellcheckerResponseForModifiedCorrection):
(WebCore::AlternativeTextController::applyDictationAlternative):
* Source/WebCore/editing/ApplyStyleCommand.cpp:
(WebCore::ApplyStyleCommand::applyInlineStyleChange):
* Source/WebCore/editing/CompositeEditCommand.cpp:
(WebCore::copyMarkers):
(WebCore::CompositeEditCommand::moveParagraphs):
* Source/WebCore/editing/DeleteSelectionCommand.cpp:
(WebCore::DeleteSelectionCommand::initializePositionData):
(WebCore::DeleteSelectionCommand::handleGeneralDelete):
(WebCore::DeleteSelectionCommand::originalStringForAutocorrectionAtBeginningOfSelection):
* Source/WebCore/editing/EditCommand.h:
(WebCore::EditCommand::protectedDocument const):
* Source/WebCore/editing/Editing.cpp:
(WebCore::enclosingEmptyListItem):
* Source/WebCore/editing/Editor.cpp:
(WebCore::Editor::updateMarkersForWordsAffectedByEditing):
(WebCore::Editor::selectionStartHasMarkerFor const):
* Source/WebCore/editing/FrameSelection.cpp:
(WebCore::FrameSelection::updateFromAssociatedLiveRange):
* Source/WebCore/editing/IndentOutdentCommand.cpp:
(WebCore::IndentOutdentCommand::tryIndentingAsListItem):
(WebCore::IndentOutdentCommand::outdentParagraph):
* Source/WebCore/editing/InsertListCommand.cpp:
(WebCore::adjacentEnclosingList):
* Source/WebCore/editing/ModifySelectionListLevel.cpp:
(WebCore::getStartEndListChildren):
(WebCore::canIncreaseListLevel):
(WebCore::IncreaseSelectionListLevelCommand::doApply):
(WebCore::IncreaseSelectionListLevelCommand::canIncreaseSelectionListLevel):
(WebCore::canDecreaseListLevel):
(WebCore::DecreaseSelectionListLevelCommand::doApply):
(WebCore::DecreaseSelectionListLevelCommand::canDecreaseSelectionListLevel):
* Source/WebCore/editing/ReplaceNodeWithSpanCommand.cpp:
(WebCore::ReplaceNodeWithSpanCommand::doUnapply):
* Source/WebCore/editing/ReplaceSelectionCommand.cpp:
(WebCore::positionAvoidingPrecedingNodes):
(WebCore::ReplaceSelectionCommand::doApply):
* Source/WebCore/editing/TextIterator.cpp:
(WebCore::TextIterator::TextIterator):
(WebCore::TextIterator::init):
(WebCore::TextIterator::advance):
(WebCore::TextIterator::handleTextNode):
(WebCore::TextIterator::handleTextRun):
(WebCore::TextIterator::handleReplacedElement):
(WebCore::TextIterator::shouldRepresentNodeOffsetZero):
(WebCore::TextIterator::representNodeOffsetZero):
(WebCore::TextIterator::handleNonTextNode):
(WebCore::TextIterator::exitNode):
(WebCore::TextIterator::protectedCurrentNode const):
(WebCore::SimplifiedBackwardsTextIterator::handleReplacedElement):
(WebCore::SimplifiedBackwardsTextIterator::handleNonTextNode):
(WebCore::SimplifiedBackwardsTextIterator::exitNode):
* Source/WebCore/editing/TextIterator.h:
(WebCore::SimplifiedBackwardsTextIterator::node const):
(WebCore::SimplifiedBackwardsTextIterator::protectedNode const):
* Source/WebCore/editing/VisibleSelection.cpp:
(WebCore::VisibleSelection::adjustSelectionRespectingGranularity):
(WebCore::VisibleSelection::adjustSelectionToAvoidCrossingEditingBoundaries):
* Source/WebCore/editing/cocoa/WebContentReaderCocoa.mm:
(WebCore::replaceRichContentWithAttachments):
* Source/WebCore/editing/markup.cpp:
(WebCore::restoreAttachmentElementsInFragment):
* Source/WebCore/page/ios/FrameIOS.mm:
(WebCore::LocalFrame::interpretationsForCurrentRoot const):
* Source/WebCore/rendering/MarkedText.cpp:
(WebCore::MarkedText::collectForDocumentMarkers):
* Source/WebCore/rendering/RenderReplaced.cpp:
(WebCore::draggedContentContainsReplacedElement):
* Source/WebCore/rendering/RenderText.cpp:
(WebCore::RenderText::draggedContentRangesBetweenOffsets const):
* Source/WebCore/testing/Internals.cpp:
* Source/WebKit/WebProcess/WebPage/Cocoa/TextCheckingControllerProxy.mm:
(WebKit::TextCheckingControllerProxy::annotatedSubstringBetweenPositions):
* Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
(WebKit::WebPage::dictationAlternativesAtSelection):
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::getPlatformEditorState const):
* Source/WebKitLegacy/ios/WebCoreSupport/WebVisiblePosition.mm:
(-[WebVisiblePosition enclosingRangeWithDictationPhraseAlternatives:]):
(-[WebVisiblePosition enclosingRangeWithCorrectionIndicator]):
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(-[WebFrame getDictationResultRanges:andMetadatas:]):

Canonical link: <a href="https://commits.webkit.org/268054@main">https://commits.webkit.org/268054@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bfe6840acad061cbd8ac656ce370c1b17439279d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/18529 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/18864 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/19469 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/20386 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/17332 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/18724 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/22177 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/19009 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/19209 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/18753 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/18928 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/16138 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/21262 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/16150 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/16900 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/23356 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/16058 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/17169 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/17070 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/21251 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/17804 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/17649 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/14959 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/21894 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/16708 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/5353 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4405 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/21073 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/23138 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/17490 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/5207 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebCore/dom/DocumentMarker.h:
```diff
@@ -126,6 +127,18 @@ class DocumentMarker {
+        , m_data(WTFMove(other.m_data))
+    { }
+
+    DocumentMarker(const DocumentMarker& other)
```
Does `DocumentMarker(const DocumentMarker& other) = default;` not work in this case?

cdumez replied:
No, because the new base class `CanMakeWeakPtr` doesn't support copying.

darinadler replied:
Seems like we could enhance `CanMakeWeakPtr` by adding an appropriate copy constructor that is the same as the default constructor.

I think we should strive to avoid unnecessary lists of data members that can fall out of date. Maybe we should start using structures in cases like this so we can take advantage of copy constructors and == operators without listing data members. New constraints might lead us to new coding idioms.

cdumez replied:
Ok, I'll add the copy constructor to CanMakeWeakPtr to avoid this.

brentfulgham reviewed (approved):
I worry that some of the really short-lived Ref's inside of loops may lead to performance issues, but I guess our bots will reveal whether it does or not.

Otherwise, this looks great! r=me

darinadler commented on Source/WebCore/editing/DeleteSelectionCommand.cpp:
```diff
@@ -303,9 +303,9 @@ bool DeleteSelectionCommand::initializePositionData()
     
     if (!isEditablePosition(start, ContentIsEditable))
-        start = firstEditablePositionAfterPositionInRoot(start, highestEditableRoot(start));
+        start = firstEditablePositionAfterPositionInRoot(start, RefPtr { highestEditableRoot(start) }.get());
```
Should we just change these functions to return `RefPtr`? What can we do with a raw pointer without getting a static analyzer warning?

darinadler commented on Source/WebCore/editing/DeleteSelectionCommand.cpp:
```diff
@@ -741,7 +741,7 @@ void DeleteSelectionCommand::handleGeneralDelete()
                     }
                     removeChildrenInRange(*m_downstreamEnd.deprecatedNode(), offset, m_downstreamEnd.deprecatedEditingOffset());
-                    m_downstreamEnd = makeDeprecatedLegacyPosition(m_downstreamEnd.deprecatedNode(), offset);
+                    m_downstreamEnd = makeDeprecatedLegacyPosition(RefPtr { m_downstreamEnd.deprecatedNode() }.get(), offset);
```
Seems like `makeDeprecatedLegacyPosition` takes ownership so it should take a `RefPtr` or `RefPtr&&`?

darinadler commented on Source/WebCore/editing/Editor.cpp:
```diff
@@ -4056,14 +4056,14 @@ static RefPtr<Node> findFirstMarkable(Node* startingNode)
 bool Editor::selectionStartHasMarkerFor(DocumentMarker::MarkerType markerType, int from, int length) const
 {
-    auto node = findFirstMarkable(m_document.selection().selection().start().deprecatedNode());
+    auto node = findFirstMarkable(RefPtr { m_document.selection().selection().start().deprecatedNode() }.get());
```
I continue to find this syntax so unpleasant I want us to refactor to avoid it. I think we can maybe change more functions to return or take `RefPtr` which makes a little extra churn when we call multiple functions on the same value but avoids this unpleasant idiom.

darinadler commented on Source/WebCore/editing/Editor.cpp:
```diff
@@ -4056,14 +4056,14 @@ static RefPtr<Node> findFirstMarkable(Node* startingNode)
     unsigned int endOffset = static_cast<unsigned int>(from + length);
-    Vector<RenderedDocumentMarker*> markers = document().markers().markersFor(*node);
-    for (auto* marker : markers) {
+    auto markers = document().markers().markersFor(*node);
```
I think we could just avoid the local variable here.

darinadler commented on Source/WebCore/editing/FrameSelection.cpp:
```diff
@@ -2949,8 +2949,8 @@ void FrameSelection::updateFromAssociatedLiveRange()
         // Don't use VisibleSelection's constructor that takes a SimpleRange, because it uses makeDeprecatedLegacyPosition instead of makeContainerOffsetPosition.
-        auto start = makeContainerOffsetPosition(&m_associatedLiveRange->startContainer(), m_associatedLiveRange->startOffset());
-        auto end = makeContainerOffsetPosition(&m_associatedLiveRange->endContainer(), m_associatedLiveRange->endOffset());
+        auto start = makeContainerOffsetPosition(Ref { m_associatedLiveRange->startContainer() }.ptr(), m_associatedLiveRange->startOffset());
```
Again this takes ownership so it seems the argument should not be a raw pointer.

darinadler commented on Source/WebCore/editing/ReplaceNodeWithSpanCommand.cpp:
```diff
@@ -69,9 +69,10 @@ void ReplaceNodeWithSpanCommand::doApply()
+    if (!spanElement || !spanElement->isConnected())
         return;
-    swapInNodePreservingAttributesAndChildren(m_elementToReplace, *m_spanElement);
+    swapInNodePreservingAttributesAndChildren(m_elementToReplace.copyRef(), *spanElement);
```
Do we really need to write `copyRef` here?

cdumez replied:
Even though `m_elementToReplace` is a smart pointer, it is not a local variable and thus may go away during the execution of the function. As a result, the static analyzer was complaining. The copyRef() makes sure we have a *local* smart pointer guaranteed to live during the execution of the function.

cdumez replied:
May look nicer to have swapInNodePreservingAttributesAndChildren() take in a `Ref<HTMLElement>`.

darinadler commented on Source/WebCore/editing/TextIterator.cpp:
```diff
@@ -586,7 +587,7 @@ bool TextIterator::handleTextNode()
             return false;
         int rendererTextLength = rendererText.length();
-        int end = (&textNode == m_endContainer) ? m_endOffset : INT_MAX;
+        int end = (textNode.ptr() == m_endContainer) ? m_endOffset : INT_MAX;
```
I wonder if we should overload `==` on `Ref` so we can do these without writing `ptr()`?

darinadler commented on Source/WebCore/editing/TextIterator.cpp:
```diff
@@ -743,28 +744,28 @@ bool TextIterator::handleReplacedElement()
 
     if (m_lastTextNodeEndedWithCollapsedSpace) {
-        emitCharacter(' ', *m_lastTextNode->parentNode(), m_lastTextNode, 1, 1);
+        emitCharacter(' ', Ref { *m_lastTextNode->parentNode() }, RefPtr { m_lastTextNode.get() }.get(), 1, 1);
```
Should we have `protectedParentNode` and maybe even `protectedNonNullParentNode` since this is so common?

darinadler commented on Source/WebCore/editing/TextIterator.cpp:
```diff
@@ -743,28 +744,28 @@ bool TextIterator::handleReplacedElement()
-            m_node = innerTextElement->containingShadowRoot();
-            pushFullyClippedState(m_fullyClippedStack, *m_node);
+            m_currentNode = innerTextElement->containingShadowRoot();
+            pushFullyClippedState(m_fullyClippedStack, *protectedCurrentNode());
```
Seems like this probably takes ownership so it should take `Ref` and caller should be doing `releaseNonNull` or have a `protectedNonNullCurrentNode`.

cdumez replied:
This doesn't take ownership, it pushes a boolean to the stack.

darinadler commented on Source/WebCore/editing/TextIterator.cpp:
```diff
@@ -743,28 +744,28 @@ bool TextIterator::handleReplacedElement()
-            pushFullyClippedState(m_fullyClippedStack, *m_node);
+    if (m_behaviors.contains(TextIteratorBehavior::EntersImageOverlays) && is<HTMLElement>(m_currentNode.get()) && ImageOverlay::hasOverlay(downcast<HTMLElement>(*m_currentNode))) {
+        if (RefPtr shadowRoot = m_currentNode->shadowRoot()) {
+            m_currentNode = shadowRoot.get();
```
Why is `get()` needed here?

cdumez replied:
You can't assign a RefPtr to a WeakPtr currently. You have to get a raw pointer from the RefPtr first.

darinadler commented on Source/WebCore/editing/TextIterator.h:
```diff
@@ -128,29 +129,29 @@ class TextIterator {
 
     // Current position, not necessarily of the text being returned, but position as we walk through the DOM tree.
-    Node* m_node { nullptr };
+    WeakPtr<Node, WeakPtrImplWithEventTargetData> m_currentNode;
```
Why `WeakPtr` instead of `RefPtr`? Weak pointer seems less efficient and not as good semantically either.

cdumez replied:
In general, we replace Raw pointers with WeakPtr or CheckedPtr. Node currently subclasses CanMakeWeakPtr but not CanMakeCheckedPtr. It is unclear if it is OK to make Node larger by subclassing both but we probably should try.

Now you're suggesting `RefPtr`. This might be possible but it is risky in general as we might introduce a reference cycle. It has to be done very carefully.

cdumez replied:
I talked to @rniwa and he thinks we should try making Node subclass CanMakeCheckedPtr.

cdumez replied:
I will look into having Node subclass CanMakeCheckedPtr in a follow-up.

darinadler replied:
Text iterator is a case that won’t introduce cycles. It’s a worker object not owned by anything; it’s basically always on the stack in a local. It should use `RefPtr`, not `WeakPtr`. Elsewhere I understand why `RefPtr` could be risky.

cdumez replied:
Ok, I'll make the change. Thanks for the clarification.

darinadler commented on Source/WebCore/editing/VisibleSelection.cpp:
```diff
@@ -294,7 +294,7 @@ void VisibleSelection::adjustSelectionRespectingGranularity(TextGranularity gran
             VisiblePosition end(wordEnd);
             
-            if (isEndOfParagraph(originalEnd) && !isEmptyTableCell(m_start.deprecatedNode())) {
+            if (isEndOfParagraph(originalEnd) && !isEmptyTableCell(RefPtr { m_start.deprecatedNode() }.get())) {
```
Since a `Position` holds a `RefPtr` to a node we should have a function that can take advantage of that instead of churning just to read a member in a case like this?

cdumez committed:
Use more smart pointers in editing node
https://bugs.webkit.org/show_bug.cgi?id=261572

Reviewed by Darin Adler and Brent Fulgham.

* Source/WebCore/dom/DocumentMarker.h:
(WebCore::DocumentMarker::DocumentMarker):
* Source/WebCore/dom/DocumentMarkerController.cpp:
(WebCore::DocumentMarkerController::markersFor):
(WebCore::DocumentMarkerController::markersInRange):
(): Deleted.
* Source/WebCore/dom/DocumentMarkerController.h:
* Source/WebCore/editing/AlternativeTextController.cpp:
(WebCore::markersHaveIdenticalDescription):
(WebCore::AlternativeTextController::respondToChangedSelection):
(WebCore::AlternativeTextController::recordSpellcheckerResponseForModifiedCorrection):
(WebCore::AlternativeTextController::applyDictationAlternative):
* Source/WebCore/editing/ApplyStyleCommand.cpp:
(WebCore::ApplyStyleCommand::applyInlineStyleChange):
* Source/WebCore/editing/CompositeEditCommand.cpp:
(WebCore::copyMarkers):
(WebCore::CompositeEditCommand::moveParagraphs):
* Source/WebCore/editing/DeleteSelectionCommand.cpp:
(WebCore::DeleteSelectionCommand::initializePositionData):
(WebCore::DeleteSelectionCommand::handleGeneralDelete):
(WebCore::DeleteSelectionCommand::originalStringForAutocorrectionAtBeginningOfSelection):
* Source/WebCore/editing/EditCommand.h:
(WebCore::EditCommand::protectedDocument const):
* Source/WebCore/editing/Editing.cpp:
(WebCore::enclosingEmptyListItem):
* Source/WebCore/editing/Editor.cpp:
(WebCore::Editor::updateMarkersForWordsAffectedByEditing):
(WebCore::Editor::selectionStartHasMarkerFor const):
* Source/WebCore/editing/FrameSelection.cpp:
(WebCore::FrameSelection::updateFromAssociatedLiveRange):
* Source/WebCore/editing/IndentOutdentCommand.cpp:
(WebCore::IndentOutdentCommand::tryIndentingAsListItem):
(WebCore::IndentOutdentCommand::outdentParagraph):
* Source/WebCore/editing/InsertListCommand.cpp:
(WebCore::adjacentEnclosingList):
* Source/WebCore/editing/ModifySelectionListLevel.cpp:
(WebCore::getStartEndListChildren):
(WebCore::canIncreaseListLevel):
(WebCore::IncreaseSelectionListLevelCommand::doApply):
(WebCore::IncreaseSelectionListLevelCommand::canIncreaseSelectionListLevel):
(WebCore::canDecreaseListLevel):
(WebCore::DecreaseSelectionListLevelCommand::doApply):
(WebCore::DecreaseSelectionListLevelCommand::canDecreaseSelectionListLevel):
* Source/WebCore/editing/ReplaceNodeWithSpanCommand.cpp:
(WebCore::ReplaceNodeWithSpanCommand::doUnapply):
* Source/WebCore/editing/ReplaceSelectionCommand.cpp:
(WebCore::positionAvoidingPrecedingNodes):
(WebCore::ReplaceSelectionCommand::doApply):
* Source/WebCore/editing/TextIterator.cpp:
(WebCore::TextIterator::TextIterator):
(WebCore::TextIterator::init):
(WebCore::TextIterator::advance):
(WebCore::TextIterator::handleTextNode):
(WebCore::TextIterator::handleTextRun):
(WebCore::TextIterator::handleReplacedElement):
(WebCore::TextIterator::shouldRepresentNodeOffsetZero):
(WebCore::TextIterator::representNodeOffsetZero):
(WebCore::TextIterator::handleNonTextNode):
(WebCore::TextIterator::exitNode):
(WebCore::TextIterator::protectedCurrentNode const):
(WebCore::SimplifiedBackwardsTextIterator::handleReplacedElement):
(WebCore::SimplifiedBackwardsTextIterator::handleNonTextNode):
(WebCore::SimplifiedBackwardsTextIterator::exitNode):
* Source/WebCore/editing/TextIterator.h:
(WebCore::SimplifiedBackwardsTextIterator::node const):
(WebCore::SimplifiedBackwardsTextIterator::protectedNode const):
* Source/WebCore/editing/VisibleSelection.cpp:
(WebCore::VisibleSelection::adjustSelectionRespectingGranularity):
(WebCore::VisibleSelection::adjustSelectionToAvoidCrossingEditingBoundaries):
* Source/WebCore/editing/cocoa/WebContentReaderCocoa.mm:
(WebCore::replaceRichContentWithAttachments):
* Source/WebCore/editing/markup.cpp:
(WebCore::restoreAttachmentElementsInFragment):
* Source/WebCore/page/ios/FrameIOS.mm:
(WebCore::LocalFrame::interpretationsForCurrentRoot const):
* Source/WebCore/rendering/MarkedText.cpp:
(WebCore::MarkedText::collectForDocumentMarkers):
* Source/WebCore/rendering/RenderReplaced.cpp:
(WebCore::draggedContentContainsReplacedElement):
* Source/WebCore/rendering/RenderText.cpp:
(WebCore::RenderText::draggedContentRangesBetweenOffsets const):
* Source/WebCore/testing/Internals.cpp:
* Source/WebKit/WebProcess/WebPage/Cocoa/TextCheckingControllerProxy.mm:
(WebKit::TextCheckingControllerProxy::annotatedSubstringBetweenPositions):
* Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
(WebKit::WebPage::dictationAlternativesAtSelection):
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::WebPage::getPlatformEditorState const):
* Source/WebKitLegacy/ios/WebCoreSupport/WebVisiblePosition.mm:
(-[WebVisiblePosition enclosingRangeWithDictationPhraseAlternatives:]):
(-[WebVisiblePosition enclosingRangeWithCorrectionIndicator]):
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
(-[WebFrame getDictationResultRanges:andMetadatas:]):

Canonical link: https://commits.webkit.org/268054@main
